### DISCUSSION TITLE
Voting/Polls support

### DISCUSSION THREAD

**Discussion by naim94a** [bluesky-social/atproto#1310] [Bluesky Lexicons]
Would be nice if Bluesky added Vote/Poll support, unlike twitter - all votes would be public (unless there's a reasonable way to keep them private).

**Comment by bnewbold** [comment]
This idea is on our radar! It is also something that 3rd parties could potentially do as an extension or independent lexicon. In fact, it would make a great demonstration of how the Lexicon system works :thinking:.

**Reply by pfrazee** [reply]
We haven't introduced the mechanisms for schema negotiation yet and we also don't have a great story around lexicon hosting, so while a 3p client could do this it might be a little premature (more difficult than if we held off).

**Reply by MarkBennett** [reply]
Would that mean the next step on moving this forward would be adding it to the AT protocol. How would someone move that forward? Is there an RFC process?

**Comment by TheTopMostDog** [comment]
I'd be stoked to see polls include options for 6 or even 8 choices, as well as the choice of check boxes (and/or/all) alongside traditional radio buttons (mutually exclusive/or-only). Being stuck with strawpoll and the like just because microblog platforms don't flesh out their polls the way they should is a shame, and less streamlined than it should be.

Oh, and give us a "view results" button instead of having to sacrifice one of the options for people who just wanna click through (to stop them clicking just anything, it's important to dedicate an option to "abstain" on platforms that don't offer show results). Then, let the poll author choose whether a user can vote after seeing the results, or if skipping forces them to abstain (useful for popularity votes).

👍👍😆

**Comment by Tamschi** [comment]
In terms of UX, it would be a good idea to have a look at WhatsApp's polls (in groups), I think.
They communicate very clearly that they aren't anonymous, and they also allow users to revoke votes.
(I'm much less of a fan of them using circular check boxes also for multiple-choice polls though.)

Here's a good walkthrough that shows everything but the detailed vote list: https://www.youtube.com/watch?v=Mg6nTim1xXk

I think that revocable votes are a necessary consequence of votes being public, and I think allowing users to reconsider should be fine too.
The AppView can still make an (anonymous) snapshot of the tallies at the preset end time, or even chart them over time in regular intervals.

**Reply by DavidBuchanan314** [reply]
Discord also does non-anonymous polls, although they don't do an amazing job of communicating the non-anonymity. There's a button to see who voted for what, which makes it pretty explicit, but the button itself isn't very visible (in classic modern interface design fashion, it doesn't even look like a button, it's just text that happens to be clickable). They also let you un-vote.

**Comment by bslade** [comment]
A long long time ago, as a techie, I implemented a guerilla write-in poll app for America Online (not budgeted or coordinated with the business managers, using spare H/W that my sys admin buddies let me use).  Poll choices were displayed with most votes at the top.  I would seed it with some initial choices.   Users would then write-in their choices and vote for them.   Usually the first page or so of choices added were the only ones that would get significant votes.   Even if really pithy choices were added later on, nobody would scroll down to look at them.   Really what's needed is a discussion/nomination time with the results of the discussion used as input by a moderator to create the initial poll.

It was fun.  Usually a snarky choice would rise to the top (eg., "Your mama" for president)

Also, even back then (2002?), some of the political polls were taken over by fervent libertarian types.  

One problem with the AOL authentication system, users could change their screen names (the unique user identifier) which allowed them to game the poll app.

Eventually, upper management (Ted Leonsis I think) found this weird unbudgeted thing and had it cancelled (Telling me me to stick to my regular work)

I'm researching the AT Protocol to see what would be involved with implementing something similar.

**Comment by keaysma** [comment]
From what I can see, this is the main place for discussion on polls? I decided to take a shot at implementing support for them in the atproto library, and put up a partially done PR here: ~https://github.com/bluesky-social/atproto/pull/3248~ https://github.com/bluesky-social/atproto/pull/3298. I'd love to get any feedback I can from anyone interested!

**Comment by xand3rrx** [comment]
I've built https://blueskypolls.xyz as a workaround for polls, feel free to give it a go - currently supports custom emoji options, end dates, hashtag support on poll questions, etc

**Reply by bslade** [reply]
Re: I've built https://blueskypolls.xyz/ as a workaround for polls,
My first reaction is, "wow, that's great".  My second reaction is, "Type my Bluesky app password into an unknown web page?  I don't think so".   I'll setup a throwaway account to test this.

**Reply by keaysma** [reply]
In defense of @xand3rrx, this is kind-of how atproto works altogether, you need to authenticate with your PDS in order to perform an action. That being said, this would probably benefit from using the oauth approach for sign in, the way [Smoke Signals](https://smokesignal.events/oauth/login) does. This has you authenticate _directly_ with your PDS instead of going through a 3rd party web interface.

I tried this out but got an error when trying to create a poll unfortunately. I'd be open to trying again in the future!

**Reply by xand3rrx** [reply]
> Re: I've built https://blueskypolls.xyz/ as a workaround for polls, My first reaction is, "wow, that's great". My second reaction is, "Type my Bluesky app password into an unknown web page? I don't think so". I'll setup a throwaway account to test this.

I too feel that a more trustworthy approach would be better for people who doesn't feel like using an app password, but as @keaysma mentioned, this is just how atproto works at the moment.

@keaysma, have you encountered any errors? I'll also work on implementing OAuth this weekend. Thanks!

**Reply by Hicclub** [reply]
Works great. Would be nice to have other poll options, like ranked choice, tho.

**Reply by QINGCHARLES** [reply]
Anyone posted one of these polls? Would like to see an example.

@xand3rrx if this is your Bluesky account then you've managed to break Bluesky :)
<img width="1158" height="546" alt="image" src="https://github.com/user-attachments/assets/cd6c7013-ebf2-4420-8ff4-1aa312715edf" />

**Comment by md-weber** [comment]
I would like to also add "Quiz" as a format, it is a feature that I really enjoy in YouTube. This is a format where you can add a question with answers, and if one of them is correct the user gets a small explanation. Great for tech content.

![image](https://github.com/user-attachments/assets/4e9a7b9b-a8c0-4014-b107-bb0b5f41a8e4)
